plugins/plugin-chart-echarts/src/Tree/controlPanel.tsx

```diff
@@ -0,0 +1,229 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { t } from '@superset-ui/core';
+import { ControlPanelConfig, sections, sharedControls } from '@superset-ui/chart-controls';
+import { DEFAULT_FORM_DATA } from './types';
+
+const requiredEntity = {
+  ...sharedControls.entity,
+  clearable: false,
+};
+
+const controlPanel: ControlPanelConfig = {
+  controlPanelSections: [
+    sections.legacyRegularTime,
+    {
+      label: t('Query'),
+      expanded: true,
+      controlSetRows: [
+        [
+          {
+            name: 'id',
+            config: {
+              ...requiredEntity,
+              label: t('Id'),
+              description: t('Name of the id column'),
+            },
+          },
+        ],
+        [
+          {
+            name: 'relation',
+            config: {
+              ...requiredEntity,
+              label: t('Relation'),
+              description: t('Name of the relation column'),
+            },
+          },
+        ],
+        [
+          {
+            name: 'name',
+            config: {
+              ...requiredEntity,
+              label: t('Name'),
+              description: t('Name of the data column'),
+            },
+          },
+        ],
```

Comment:
I think it will be typical to only have string-based `id` and `parent` fields, in which case the `name` control would be redundant. Perhaps we could make this one optional and add a note about it in the description?